hazelcast/src/hazelcast/client/proxy.cpp

```diff
@@ -76,6 +76,15 @@ namespace hazelcast {
             return true;
         }
 
+        void reliable_topic::on_shutdown() {
+            // cancel all runners
+            for (auto &entry : runners_map_.clear()) {
+                entry.second->cancel();
+            }
+            // destroy the underlying ringbuffer
+            ringbuffer_.get()->destroy().get();
```

Comment:
`DistributedObject::destroy` destroys the object cluster-wide. Won't this have effects on other clients and members using the same reliable topic?

Reply:
This ringbuffer is internal special ringbuffer for this topic but yes, if another client is using the same ringbuffer, that would cause a problem. i will remove this cleanup responsibility to the user to call destroy explicitly on the topic to achieve this cleanup. Removing this line.